### PR TITLE
Configurable possibility to channel on GoPay

### PR DESCRIPTION
**Pull request by martinsifra**
Now you can configure (changeChannel config parameter) a possibility to
allow or deny a change of a payment channel on GoPay gate.

### CODE REVIEW

**Comment by f3l1x** [comment]
I like it, but only as property and settter. Could you change it please?

**Comment by martinsifra** [comment]
You think without setting up from config?

**Comment by f3l1x** [comment]
@martinsifra I could be over config, but via setter method.

For example like this.

```php
if ($config[...]) {
$class->addSetup('setChangeChannel', TRUE);
}
```

**Comment by martinsifra** [comment]
Yeah, I see. I'll make commit today.

**Comment by f3l1x** [comment]
ping @martinsifra

**Comment by martinsifra** [comment]
Sorry, I know. It is on my schedule.

File: src/Gopay/Service.php [hunk 1]
```diff
@@ -183,6 +186,12 @@ public function setTestMode($testMode = TRUE)
 		GopayConfig::$version = $this->testMode ? GopayConfig::TEST : GopayConfig::PROD;
 		return $this;
 	}
+	
+	public function setChangeChannel($changeChannel = TRUE)
```
Comment (f3l1x) [inline src/Gopay/Service.php]: indentation

File: src/Gopay/Extension.php [hunk 2]
```diff
@@ -30,6 +30,7 @@ class Extension extends CompilerExtension
 		'gopayId' => NULL,
 		'gopaySecretKey' => NULL,
 		'testMode' => TRUE,
+		'changeChannel' => TRUE,
```
Comment (f3l1x) [inline src/Gopay/Extension.php]: better default NULL

File: src/Gopay/Extension.php [hunk 3]
```diff
@@ -46,8 +47,10 @@ public function loadConfiguration()
 				$driver,
 				$config['gopayId'],
 				$config['gopaySecretKey'],
-				isset($config['testMode']) ? $config['testMode'] : FALSE
+				isset($config['testMode']) ? $config['testMode'] : FALSE,
 			));
+			
+		$service->addSetup('setChangeChannel', array($config['changeChannel']));
```
Comment (f3l1x) [inline src/Gopay/Extension.php]: add setup only if changeChannel is filled (TRUE or FALSE)

**Comment by f3l1x** [comment]
@martinsifra It's great. Could you prepare simple test? And primary, rebase it to 1 commit?

**Comment by martinsifra** [comment]
Thanks, I could, but I've never do that. So I'll need more time :-)

**Comment by f3l1x** [comment]
@martinsifra It's oukey. I'll do it by my own.

**Comment by f3l1x** [comment]
Merged via https://github.com/Markette/Gopay/commit/d545d5da6b7d7523ae903a10c5716d9e4c863534. Thank you.

**Pull request closed without merging**